### DISCUSSION TITLE
General method of contributing pre-existing plugin code

### DISCUSSION THREAD

**Discussion by andrew-lyons** [capawesome-team/capacitor-mlkit#29] [Q&A] [answered]
Hello!

I have a few MLKit plugins that we've built for inter-company use that we wanted to open source. Thankfully this project came along!

We built this with a specific Ionic/Capacitor version in mind, as well was with company specific conventions, etc. I was wondering what the preferred process would look like for contributing a PR for one of our plugins (Digital Ink). Specifically, I had questions around:

- Ionic/Cap versions, are we just supporting newest versions etc?
- Test ionic apps. It doesn't look like these are included in this project, should we just spin up a generic app or provide test application code since some of these mlkit models have some scaffolding required in order to perform inference
- Unit testing

I look forward to contributing, thanks for the help.

**Comment by robingenz** [comment, marked as answer]
Hi @andrew-lyons,
your contribution would be great and am of course happy to help!

Regarding your questions:

> Ionic/Cap versions, are we just supporting newest versions etc?

Yes, we always support the latest Capacitor version. I have also already updated both plugins in this repo to Capacitor 5 (see #28 and #31).

>  Test ionic apps. It doesn't look like these are included in this project, should we just spin up a generic app or provide test application code since some of these mlkit models have some scaffolding required in order to perform inference

There is already a demo app, just in a separate project: https://github.com/robingenz/capacitor-mlkit-plugin-demo

> Unit testing

Unit tests are currently not yet a requirement. But you can of course add some if you like.

**My suggestion**: Let me know which ML Kit plugins you want to add and I will prepare the boilerplate code for you, so you only need to add the implementation. This will also make the review easier.
It's also important that the source code is as clean and consistent as possible with the other plugins, so that anyone who wants to contribute can quickly find their way around. Unfortunately, I don't have specific guidelines for this yet. You can have a look at my other plugins to get an idea.

**Reply by andrew-lyons** [reply]
Thanks. That sounds like a plan. Let's start with the Digital Ink plugin, since I already have an implementation working that I'll just need to update for the newest capacitor version. I'll stay tuned til you get the boilerplate ready, in the meantime will get a sense for how the other plugins are built. :+1:

**Reply by robingenz** [reply]
Great, I will set everything up in the next few days and then let you know.

**Reply by robingenz** [reply]
@andrew-lyons Sorry, I totally forgot about this discussion. 🤦‍♂️ Are you still interested? If so, I'll create the boilerplate code for you next week.

**Reply by andrew-lyons** [reply]
@robingenz all good. Yes, if you are able to I would still love to contribute. I think understanding the idea around how these could be implemented could be helpful in implementing other models as well.

Just let me know :+1:

**Reply by robingenz** [reply]
@andrew-lyons I have prepared everything. You can just check out my branch and create a PR: https://github.com/capawesome-team/capacitor-mlkit/tree/feat/digital-ink-recognition

**Reply by robingenz** [reply]
Please just let me know if you have any questions. Thanks for your contribution.
